underfs/swift/src/main/java/tachyon/underfs/swift/direct/SwiftDirectOutputStream.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.underfs.swift.direct;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+
+/**
+ * Swift output stream implements OutputStream.
+ * This class is used to write data into Swift
```

Comment:
I was thinking the scenario that multiple threads are accessing the same `SwiftDirectOutputStream`, e.g., two threads concurrently are calling `write()`. In this case, I guess the class is not thread-safe. 
I don't think adding `synchronize` is necessary. As long as we specify explicitly the thread-safety, users of this class can have their own thread-coordination scheme.